### PR TITLE
sidebar-fix

### PR DESCRIPTION
**Pull request by oguzzkilic**
fixed issue #630 
sidebar-fix-yeoman

### CODE REVIEW

**Comment by SBoudrias** [comment]
How is this acting on mobile?

File: app/assets/js/script.js [hunk 1]
```diff
@@ -8,6 +8,35 @@
     var $doc = $(document);
     var $body = $(document.body);
 
+    (function() {
```
Comment (SBoudrias) [inline app/assets/js/script.js]: No need or this extra scope.

**Comment by oguzzkilic** [comment]
@SBoudrias this section is not use on mobile.

**Comment by oguzzkilic** [comment]
Is there anyone there ? @eddiemonge @SBoudrias

File: app/assets/css/main.css [hunk 2]
```diff
@@ -607,7 +607,12 @@ Context/Local Nav
 }
 .context-nav.navbar-fixed {
   position: fixed;
-  top: 7.7em;
+  top: 6em;
+}
+
+.context-nav.navbar-absolute {
+  position: absolute;
+  right: 0!important;
```
Comment (SBoudrias) [inline app/assets/css/main.css]: Can you find a way to not use `!important`?

File: app/assets/js/script.js [hunk 3]
```diff
@@ -8,6 +8,32 @@
...
 
+    var fixedElementOffset = $('.context-nav').offset().top;
+    var footerOffset = $('.SiteFooter').offset().top - 93;
+    var fixedElementHeight = $('.context-nav').height();
+
+    $(window).scroll(function (event) {
+      var y = $(this).scrollTop();
+
+      if (y < 148) {
+        $('.context-nav').css('top','1.3em');
```
Comment (SBoudrias) [inline app/assets/js/script.js]: Let's cache all `$()` selectors outside the closure so we don't need to recompute them for every event triggered during scroll.

Comment (SBoudrias) [inline app/assets/js/script.js]: Also, please add a space after each `,`

File: app/assets/js/script.js [hunk 4]
```diff
@@ -8,6 +8,32 @@
...
+      } else {
+        $('.context-nav').css('top','6em');
+      }
+
+      if ( y >= fixedElementOffset && ( y + fixedElementHeight ) < footerOffset ) {
```
Comment (SBoudrias) [inline app/assets/js/script.js]: `if (exp)` (no space inside parens)

File: app/assets/js/script.js [hunk 5]
```diff
@@ -8,6 +8,32 @@
...
+        $('.context-nav').addClass('navbar-fixed').removeClass('navbar-absolute');
+        $('.context-nav').css('top','6em');
+      }
+      
+      else if ( y >= fixedElementOffset && ( y + fixedElementHeight ) >= footerOffset ) {
```
Comment (SBoudrias) [inline app/assets/js/script.js]: no need to wrap `(y + fixedEl..)` in parens.

File: app/assets/js/script.js [hunk 6]
```diff
@@ -8,6 +8,32 @@
...
+        var fixEl = (fixedElementHeight / 2) + fixedElementHeight;
+        var newOffset = Math.round($('.SiteFooter').offset().top - fixEl);
```
Comment (SBoudrias) [inline app/assets/js/script.js]: No need to round this number.

**Comment by oguzzkilic** [comment]
All fixed. @SBoudrias thanks.

**Comment by SBoudrias** [comment]
Thanks!

**Pull request merged**